package.json

```diff
@@ -34,9 +33,10 @@
   "license": "MIT",
   "dependencies": {
     "@babel/runtime": "7.7.7",
+    "@date-fns/upgrade": "1.0.1",
```

Comment:
I don't think this package needs to be installed in the project as we can run it with `npx`. But if you wish to keep it for convenience, can you move it to be a devDependency, please?

Reply:
The codemode tool ([date-fns-upgrade-codemod](https://github.com/date-fns/date-fns-upgrade-codemod)) can be run using `npx` which does not require us to add additional dependencies. Unfortunately, transitional methods provided by support package [date-fnd-upgrad](https://github.com/date-fns/date-fns-upgrade), such as `legacyParse` and `convertToken` , require us to add the package as a main dependency.

I have worked to remove the usages for `legacyParse` in this PR, but I am leaning towards keeping `convertToken`.

In v2.0, date-fns introduced breaking changes to which format strings they accept. Some changes include accepting `yyyy' instead of 'YYYY', and 'dd' instead of 'DD'. In addition, characters are now escaped using single quote symbols (`'`) instead of square brackets. See [corresponding section in the changelog](https://github.com/date-fns/date-fns/blob/master/CHANGELOG.md#200---2019-08-20)

I have not been able to find a way to remove `convertToken` and still pass the tests..

Considering the new format string validations are more strict than before, and the format is different, I am inclined to keep `convertToken` to avoid introducing breaking changes for anyone using this library.